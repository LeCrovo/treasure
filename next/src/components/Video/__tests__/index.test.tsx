import { render } from '@testing-library/react'

import VideoPlayerHtml5 from '../index'

describe('<VideoPlayerHtml5 />', () => {
  const mockVideoProps = {
    extension: 'mp4',
    poster: 'jpg',
    src: 'video.mp4',
    description: 'Sample HTML5 video',
  }
  test('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(<VideoPlayerHtml5 {...mockVideoProps} />)
    expect(spy).not.toHaveBeenCalled()
  })

  test('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<VideoPlayerHtml5 {...mockVideoProps} />)
    expect(firstChild).toMatchSnapshot()
  })
})
