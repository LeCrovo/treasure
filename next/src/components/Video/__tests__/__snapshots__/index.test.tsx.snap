// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPlayerHtml5 /> Should render and match the snapshot 1`] = `
<span
  class="image-gallery-description"
>
  Sample HTML5 video
</span>
`;
